### PR TITLE
Docker container

### PR DESCRIPTION
**Pull request by AaronDewes**
The docker images build, but haven't been tested yet.

---

This adds very early docker container support. Once you merge this PR, builds will automatically happen on every push and tag and will be pushed to GitHub container registry.

### CODE REVIEW

File: .github/workflows/on-push.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build on push
+
+permissions:
+  packages: write
+
+on:
+    push:
+        branches:
+            - main
```
[the remark this answers is not in the document]
Comment (bumi) [reply .github/workflows/on-push.yml:9]: if we want to keep building on push, then we must change "main" to "master" (we still use master as the master/main branch)

**Review by bumi — COMMENTED** [review]
How are configuration options passed in? that would be done when running the container? e.g. through docker-compose?

File: .github/workflows/on-push.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (AaronDewes) [reply .github/workflows/on-push.yml:9]: It builds on both master and main, so it still works if you rename

**Comment by AaronDewes** [comment]
Yes, options get passed in as env vars through docker-compose

File: .github/workflows/on-push.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bumi) [reply .github/workflows/on-push.yml:9]: ah I missed the `master` in the line below. sorry. thanks.

[the remark this answers is not in the document]
Comment (bumi) [reply .github/workflows/on-push.yml:9]: (though I have no plans to rename the branch and potentially break things.)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN go install
+
+RUN rice embed-go && go build
+
+FROM alpine:3.14
+
+COPY --from=builder /app/lnme /lnme
+
+EXPOSE 1323
```
Comment (bumi) [inline Dockerfile:18]: the PORT can be configured... should this then also be configurable (based on the LNME_LISTEN env variable) or is it a practice to NOT change the default port when using docker? 
(I am not very experienced with docker)

Comment (junderw) [reply Dockerfile:18]: EXPOSE just exposes a port to other containers by default.

At runtime / in a higher level compose file you can expose and/or open-to-host any port you want.

This should be fine to use the default port. (It's semi-convention)

Comment (bumi) [reply Dockerfile:18]: thanks! 👍

Comment (AaronDewes) [reply Dockerfile:18]: From the docker documentation



> The `EXPOSE` instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published.

File: .github/workflows/on-push.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (AaronDewes) [reply .github/workflows/on-push.yml:9]: Actually, branch could be left out so it also builds for WIP feature branches for testing. What do you think?

**Comment by bumi** [comment]
OK, I think this is cool. 

I can then also do something like: `docker run -it --rm ghcr.io/bumi/lnme:master` correct?
could you add a note to the readme?

**Comment by AaronDewes** [comment]
Sorry, I missed your comment. I'll add a note now.

**Comment by bumi** [comment]
thanks so much! merging!

**Pull request merged**